### PR TITLE
Feature review

### PR DESCRIPTION
**Pull request by mrkiura**
Checkpoint 1.

### CODE REVIEW

File: amity.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                line = line.strip()
+                if line:
+                    details = line.split()
+                    occupant1 = Occupant(name=details[0] + ' ' + details[1],
+                                         job_type=details[2])
+                    if details[2] == 'FELLOW' and details[-1] == 'Y':
+                        self.boarding_fellows.append(occupant1)
+                    elif details[-1] == 'N':
+                        self.non_boarding_fellows.append(occupant1)
+                    self.staff.append(occupant1)
```
Comment (afropolymath) [inline amity.py]: **Code Functionality: Nitpick**

Is everyone being appended to the staff list?

File: amity.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    def allocate(self):
+        '''
+        Calls the methods to randomly allocate
+        rooms to people
+        '''
+        # give staff offices that are closer together
+        available_staff_offices = self.get_available_rooms('OFFICE')
+        self.allocate_rooms(available_staff_offices, self.staff)
+        # give fellows offices that are closer together
+        available_fellow_offices = self.get_available_rooms('OFFICE')
```
Comment (afropolymath) [inline amity.py]: **Code Quality: Nitpick**

You made multiple calls to the `get_available_rooms` method with the same parameter.

File: amity.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        next_person = Occupants.next(people)
+        person = None
+        for room in rooms:
+            while room.is_not_full():
+                try:
+                    person = next_person.next()
+                    room.add_occupant(person)
+                except StopIteration:
+                    break
+
```
Comment (afropolymath) [inline amity.py]: **Code Quality: Critical**

You need to add a second level check to ensure when there are no more people, the iteration through rooms stops.

File: amity.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    def get_available_rooms(self, usage):
+        '''
+        Returns a list of rooms based on usage
+        e.g, returns a list of office spaces
+        if usage == 'OFFICE'
+        '''
+        available = []
+        if usage == 'OFFICE':
+            office_gen = (office for office in
```
Comment (afropolymath) [inline amity.py]: **Code Quality: Nitpick**

No need to use a generator here since you're eventually converting to a list.

File: amity.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                          self.office_rooms if office.is_not_full())
+            for i in office_gen:
```
Comment (afropolymath) [inline amity.py]: **Code Quality: Nitpick**

You would have just used the `list` function to convert the generator object to a list.

File: amity.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                available.append(i)
```
Comment (afropolymath) [inline amity.py]: **Code Quality: Nitpick**

I feel like the function should have exited at this point.

File: tests/test_amity.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    def setUp(self):
+        self.building = Amity()
+
+    def test_populate_offices(self):
+        '''
+        Test whether the populate method
+        populates the building with office rooms
+        '''
+        self.building.populate()
+        self.assertTrue(len(self.building.office_rooms) > 0)
```
Comment (afropolymath) [inline tests/test_amity.py]: **Code Quality: Nitpick**

Would be nice to ensure that the `office_rooms` list is empty before the `populate` function is called.

File: tests/test_amity.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        populates the building with living rooms
+        '''
+        self.building.populate()
+        self.assertTrue(len(self.building.living_rooms) > 0)
+
+    def test_create_staff(self):
+        '''
+        Test create_occupants method
+        '''
+        self.assertTrue(len(self.building.staff))
```
Comment (afropolymath) [inline tests/test_amity.py]: **Code Quality: Critical**

Assertion here is not well defined. You probably meant `assertTrue(len(self.building.staff) > 0)`.

File: tests/test_amity.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+    def setUp(self):
+        self.building = Amity()
+
+    def test_populate_offices(self):
+        '''
+        Test whether the populate method
+        populates the building with office rooms
+        '''
+        self.building.populate()
```
Comment (afropolymath) [inline tests/test_amity.py]: **Code Quality: Nitpick**

This is already called in the constructor. No need for this.

File: tests/test_amity.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        self.assertTrue(len(self.building.office_rooms) > 0)
+
+    def test_populate_living(self):
+        '''
+        Test whether the populate method
+        populates the building with living rooms
+        '''
+        self.building.populate()
+        self.assertTrue(len(self.building.living_rooms) > 0)
```
Comment (afropolymath) [inline tests/test_amity.py]: **Code Quality: Nitpick**

Test cases should be more exact as opposed to asserting greater than 0 e.g in this case `assertEqual(len(self.building.living_rooms), 5)`.

File: tests/test_occupant_gen.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import unittest
+from occupant_gen import Occupants
+
+
+class OccupantGenTest(unittest.TestCase):
```
Comment (afropolymath) [inline tests/test_occupant_gen.py:5]: **Code Quality: Awesome**

Great stuff!

File: amity.py (hunk 1 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: amity.py (hunk 2 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: amity.py (hunk 3 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: amity.py (hunk 4 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: amity.py (hunk 5 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: amity.py (hunk 6 above)
Comment (mrkiura) [inline amity.py]: Fixed

File: tests/test_amity.py (hunk 7 above)
Comment (mrkiura) [inline tests/test_amity.py]: Fixed

File: tests/test_amity.py (hunk 8 above)
Comment (mrkiura) [inline tests/test_amity.py]: Fixed

File: tests/test_amity.py (hunk 9 above)
Comment (mrkiura) [inline tests/test_amity.py]: Fixed

File: tests/test_amity.py (hunk 10 above)
Comment (mrkiura) [inline tests/test_amity.py]: Fixed

**Pull request merged**